### PR TITLE
Inline webpack

### PR DESCRIPTION
**Pull request by SimonBiggs**

### CODE REVIEW

**Comment by SimonBiggs** [comment]
Next step:

https://github.com/SimonBiggs/scriptedforms/issues/233

**Comment by SimonBiggs** [comment]
@saulshanabrook would I be able to get you to do a quick review of this?

File: scriptedforms/src/app/variables-module/variable-table.component.ts [hunk 1]
```diff
@@ -41,10 +39,14 @@ import { PandasTable } from '../interfaces/pandas-table';
 
 import { VariableParameterComponent } from './variable-parameter.component';
 
+import * as htmlTemplate from 'html-loader!./variable-table.component.html';
+const template = '' + htmlTemplate;
```
Comment (saulshanabrook) [inline scriptedforms/src/app/variables-module/variable-table.component.ts]: What's this for?

**Comment by SimonBiggs** [comment]
For some reason the angular linter wasn't picking up the html template as a
string. ... So... It was my quick attempt to make the linter be quiet.

I'll have to work out a nicer way.

**Comment by SimonBiggs** [comment]
Thanks Saul :)

**Comment by saulshanabrook** [comment]
Oh, odd. I would think it could infer that from the declaration file, but I guess it is getting confused somewhere.

**Comment by SimonBiggs** [comment]
Yeah. I might need to look into the linter code itself. Might need an
upstream fix.

**Comment by SimonBiggs** [comment]
@saulshanabrook unfortunately it completely breaks the linting of the angular template file. In its standard form the angular language service recognises that the template file and component are linked and it uses the typescript file to lint what is written within the html template file.

So, this method works, but the original pull request is still significantly preferable.

**Comment by SimonBiggs** [comment]
https://angular.io/guide/language-service

**Comment by SimonBiggs** [comment]
Unfortunately I don't know where to begin on providing a pull request to angular to allow interpreting imports as I've done it as a html template :/

https://github.com/angular/angular/tree/master/packages/language-service/src

**Comment by SimonBiggs** [comment]
And the template linter error wasn't a type error (hence why being a string didn't fix it) it's the angular language service not finding the template itself. By making '' + something be the "template" it stops complaining because it found a template of ''.

**Comment by SimonBiggs** [comment]
What if I tried using rollup? Might I have success down that path?

File: scriptedforms/src/app/variables-module/variable-table.component.ts (hunk 1 above)
Comment (SimonBiggs) [reply scriptedforms/src/app/variables-module/variable-table.component.ts]: For some reason the angular linter wasn't picking up the html template as a string. ... So... It was my quick attempt to make the linter be quiet. I'll have to work out a nicer way.

Comment (SimonBiggs) [reply scriptedforms/src/app/variables-module/variable-table.component.ts]: It's actually got nothing to do with string or not. It's the Angular language service not recognising the template import. (Understandably as it is quite nonstandard).

**Comment by SimonBiggs** [comment]
@saulshanabrook alright, I've created an initial issue over at Angular. I'll let you know how I progress.

https://github.com/angular/angular/issues/23478

Once that is sorted out, I'll sort out https://github.com/SimonBiggs/scriptedforms/issues/233, and then I might make an Angular JupyterLab extension cookie cutter?

**Comment by SimonBiggs** [comment]
@robmarkcole by the way, this pull requests makes jupyterlab work with the current pypi jupyterlab release. I'm just going to implement a few usability things and then ScriptedForms should be a fully fledged jupyterlab extension. :)

**Comment by robmarkcole** [comment]
Excellent will try it out this week. The acid test is does it run on my windows machine at work :)

**Pull request merged**